Mathlib/Topology/Order/LawsonTopology.lean

```diff
@@ -0,0 +1,243 @@
+/-
+Copyright (c) 2023 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+
+import Mathlib.Topology.Order.LowerUpperTopology
+import Mathlib.Topology.Order.ScottTopology
+
+/-!
+# Lawson topology
+
+This file introduces the Lawson topology on a preorder.
+
+## Main definitions
+
+- `Topology.lawson` - the Lawson topology is defined as the meet of `Topology.lower` and
+  `Topology.scott`.
+- `Topology.IsLawson.lawson_basis` - The complements of the upper closures of finite sets
+  intersected with Scott open sets.
+
+## Main statements
+
+- `Topology.IsLawson.isTopologicalBasis` - `Topology.IsLawson.lawson_basis` is a basis for
+  `Topology.IsLawson`
+- `Topology.lawsonOpen_iff_scottOpen_of_isUpperSet'` - An upper set is Lawson open if and only if it
+  is Scott open
+- `Topology.lawsonClosed_iff_dirSupClosed_of_isLowerSet` - A lower set is Lawson closed if and only
+  if it is closed under sups of directed sets
+- `Topology.IsLawson.t0Space` - The Lawson topology is T₀
+
+## Implementation notes
+
+A type synonym `Topology.WithLawson` is introduced and for a preorder `α`, `Topology.WithLawson α`
+is made an instance of `TopologicalSpace` by `Topology.lawson`.
+
+We define a mixin class `Topology.IsLawson` for the class of types which are both a preorder and a
+topology and where the topology is `Topology.lawson`.
+It is shown that `Topology.WithLawson α` is an instance of `Topology.IsLawson`.
+
+## References
+
+* [Gierz et al, *A Compendium of Continuous Lattices*][GierzEtAl1980]
+
+## Tags
+Lawson topology, preorder
+-/
+
+open Set
+
+open TopologicalSpace
+
+variable {α β : Type*}
+
+namespace Topology
+
+/-! ### Lawson topology -/
+
+section Lawson
+section Preorder
+
+/--
+The Lawson topology is defined as the meet of `Topology.lower` and the `Topology.scott`.
```

Comment:
Not very grammatical

Reply:
I've changed it to "the Lawson topology is defined as the meet of the lower topology and the Scott topology."